### PR TITLE
Feature/multisample input phasing

### PR DESCRIPTION
**Pull request by justinjj24**
- Rewrite the `main.nf` to support multisample input config and compatability of single sample input format of legacy version.
- Module based methods introduced for each process in this version.

### CODE REVIEW

**Comment by justinjj24** [comment]
test run completed. @mhebrard can merge to `develop`

**Comment by mhebrard** [comment]
Runing on my local machine I got 

```
❯ sh run.sh
N E X T F L O W  ~  version 22.04.5
WARN: It appears you have never run this project before -- Option `-resume` is ignored
Launching `../../main.nf` [tiny_faggin] DSL2 - revision: f6075d786e
N E X T F L O W  ~  version 22.04.5 5708
NPM-sample-qc  ~  version 0.9.0
User name    : hebrardms
Command Line : nextflow run ../../main.nf -config ../../nextflow.config -params-file params.yml -work-dir ./work --publish_dir ./output -resume
Project Dir  : /Users/hebrardms/Documents/Github/NPM-sample-qc
Launch Dir   : /Users/hebrardms/Documents/Github/NPM-sample-qc/tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6
Work Dir     : /Users/hebrardms/Documents/Github/NPM-sample-qc/tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/work
Results Dir  : ./output
Info Dir     : ./output/pipeline_info
Profile      : standard
Is a directory

 -- Check script '/Users/hebrardms/Documents/Github/NPM-sample-qc/main.nf' at line: 85 or see '.nextflow.log' file for more details
```

@justinjj24 can you confirm which version of Nextflow you are using for your test ?

File: conf/resources.config [hunk 1]
```diff
@@ -10,4 +10,5 @@ params {
     vbi2_bed = "${projectDir}/resources/verifybamid/1000g.phase3.100k.b38.vcf.gz.dat.bed"
     vbi2_mean = "${projectDir}/resources/verifybamid/1000g.phase3.100k.b38.vcf.gz.dat.mu"
     autosomes_non_gap_regions = "${projectDir}/resources/autosomes_non_gap_regions.bed"
+    //autosomes_non_gap_regions = "https://github.com/c-BIG/NPM-sample-qc/raw/v${manifest.version}/resources/autosomes_non_gap_regions.bed"
```
Comment (mhebrard) [inline conf/resources.config:13]: to remove this line as it is commented out

File: tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/run.sh [hunk 2]
```diff
@@ -10,4 +10,5 @@ nextflow run ../../main.nf \
 	-config ../../nextflow.config \
 	-params-file params.yml \
 	-work-dir ./work \
-	--outdir ./output
+	--publish_dir ./output \
+	-resume
```
Comment (mhebrard) [inline tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/run.sh:14]: to remove resume option

Comment (justinjj24) [reply tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/run.sh]: `-resume` option removed

File: conf/resources.config (hunk 1 above)
Comment (justinjj24) [reply conf/resources.config]: Done

**Comment by justinjj24** [comment]
> Runing on my local machine I got
> 
> ```
> ❯ sh run.sh
> N E X T F L O W  ~  version 22.04.5
> WARN: It appears you have never run this project before -- Option `-resume` is ignored
> Launching `../../main.nf` [tiny_faggin] DSL2 - revision: f6075d786e
> N E X T F L O W  ~  version 22.04.5 5708
> NPM-sample-qc  ~  version 0.9.0
> User name    : hebrardms
> Command Line : nextflow run ../../main.nf -config ../../nextflow.config -params-file params.yml -work-dir ./work --publish_dir ./output -resume
> Project Dir  : /Users/hebrardms/Documents/Github/NPM-sample-qc
> Launch Dir   : /Users/hebrardms/Documents/Github/NPM-sample-qc/tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6
> Work Dir     : /Users/hebrardms/Documents/Github/NPM-sample-qc/tests/NA12878-chr14-AKT1_1000genomes-dragen-3.7.6/work
> Results Dir  : ./output
> Info Dir     : ./output/pipeline_info
> Profile      : standard
> Is a directory
> 
>  -- Check script '/Users/hebrardms/Documents/Github/NPM-sample-qc/main.nf' at line: 85 or see '.nextflow.log' file for more details
> ```
> 
> @justinjj24 can you confirm which version of Nextflow you are using for your test ?
> 
> I see in the `nextflow_config` that the minimal nexflow version as changed from v22.x to v23.x, but runing the script did not complain that my local version is lower than the requirements...

@mhebrard Yes, please update the version as per the `nextflow.config` `nextflowVersion = '>=23.04.1'`. Lower version will not support some of the functions defined in the `main.nf`

**Review by mhebrard — APPROVED** [review]

**Pull request merged**